### PR TITLE
UPSTREAM: <carry>: Add a Dockerfile that compatible with OpenShift

### PR DESCRIPTION
**Pull request by Fedosin**

### CODE REVIEW

**Comment by lobziik** [comment]
WDYT about store openshift related things in separate directory, like `openshift-hack` like we have in azure repo?
https://github.com/openshift/cloud-provider-azure/tree/master/openshift-hack

**Review by elmiko — APPROVED** [review]
tried this out locally, worked for me
/lgtm

**Comment by Fedosin** [comment]
```sh
❯ podman build -t quay.io/mfedosin/alibaba-cloud-controller-manager -f ./openshift-hack/images/Dockerfile.openshift .
STEP 1: FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.16-openshift-4.9 AS builder
STEP 2: WORKDIR /build
--> Using cache fe02d554e53d7c46b72c010c3100ec0f0c4fab51d36274041fc6fc5040a0920a
--> fe02d554e53
STEP 3: COPY . .
--> Using cache 9964debfe68dcd1fb1f008dfdac5073e3711d16dcaccc231b97b196e6fe2d5a1
--> 9964debfe68
STEP 4: RUN make cloud-controller-manager
--> Using cache f6df2135d3708625b9fe02cc40d0d3cc3c0ceb2e9675fd044c69d9b45641e9b1
--> f6df2135d37
STEP 5: FROM registry.ci.openshift.org/ocp/4.9:base
STEP 6: LABEL description="Alibaba Cloud Controller Manager"
--> Using cache 734a47265722dabb04b3679a590fb581ada73375822d18790b7230e84c717c51
--> 734a4726572
STEP 7: COPY --from=builder /build/alibaba-cloud-ccm /bin/alibaba-cloud-ccm
--> Using cache 03fae8db9121dcfd7e1953b008e9ee14783699f3f7a178bbe710648942c40123
--> 03fae8db912
STEP 8: ENTRYPOINT [ "/bin/alibaba-cloud-ccm" ]
--> Using cache d09570435d85712c7f7cd18380176e0fa8283a1e8d45939300c4f4e27250a595
STEP 9: COMMIT quay.io/mfedosin/alibaba-cloud-controller-manager
--> d09570435d8
Successfully tagged quay.io/mfedosin/alibaba-cloud-controller-manager:latest
d09570435d85712c7f7cd18380176e0fa8283a1e8d45939300c4f4e27250a595
```

File: openshift-hack/images/Dockerfile.openshift [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+COPY . .
+RUN make cloud-controller-manager
+
+FROM registry.ci.openshift.org/ocp/4.9:base
+
+LABEL description="Alibaba Cloud Controller Manager"
+
+COPY --from=builder /build/alibaba-cloud-ccm /bin/alibaba-cloud-ccm
+
+ENTRYPOINT [ "/bin/alibaba-cloud-ccm" ]
```
Comment (JoelSpeed) [inline openshift-hack/images/Dockerfile.openshift:14]: Elsewhere, we have the binary called `platform-cloud-controller-manager`, would be good to be consistent. Is this something that's in the upstream? Can we rename it in the dockerfile if so?

Comment (Fedosin) [reply openshift-hack/images/Dockerfile.openshift:14]: sure, will be done!

**Comment by JoelSpeed** [comment]
Looks good, thanks @Fedosin

**Pull request merged**